compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -524,103 +524,98 @@ TR::Register *OMR::X86::TreeEvaluator::tableEvaluator(TR::Node *node, TR::CodeGe
+   auto result = cg->allocateRegisterPair(cg->allocateRegister(), cg->allocateRegister());
+
+   generateRegRegInstruction(CMP4RegReg, node, oprand0->getLowOrder(), oprand1->getLowOrder(), cg);
+   generateRegInstruction(SETcc, node, result->getLowOrder(), cg); // t1 = (low0 < low1)
+   generateRegRegInstruction(CMP4RegReg, node, oprand0->getHighOrder(), oprand1->getHighOrder(), cg);
+   generateRegInstruction(SETcc, node, result->getHighOrder(), cg); // t2 = (high0 < high1)
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getHighOrder(), result->getLowOrder(), cg); // if (high0 == high1) then t2 = t1 = (low0 < low1)
+
+   generateRegRegInstruction(TEST1RegReg,  node, result->getHighOrder(), result->getHighOrder(),  cg);
+   generateRegRegInstruction(MOV4RegReg,   node, result->getLowOrder(),  oprand0->getLowOrder(),  cg);
+   generateRegRegInstruction(MOV4RegReg,   node, result->getHighOrder(), oprand0->getHighOrder(), cg);
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getLowOrder(),  oprand1->getLowOrder(),  cg);
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getHighOrder(), oprand1->getHighOrder(), cg);
+   node->setRegister(result);
+   for (int i = 0; i < node->getNumChildren(); i++)
       {
-      generateRegRegInstruction(cmp_op, node, reg, trgReg, cg);
-      generateLabelInstruction(isMin ? JG4 : JL4, node, doneLabel, false, cg);
-      generateRegRegInstruction(move_op, node, trgReg, reg, cg);
-
-      dep = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 2, cg->trMemory());
-      dep->addPostCondition(trgReg, TR::RealRegister::NoReg, cg);
-      dep->addPostCondition(reg, TR::RealRegister::NoReg, cg);
+      cg->decReferenceCount(node->getChild(i));
       }
-
-   generateLabelInstruction(LABEL, node, doneLabel, dep, cg);
-
-   node->setRegister(trgReg);
+   return result;
+   }
+TR::Register *OMR::X86::TreeEvaluator::minmaxEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   if (TR::Compiler->target.is32Bit() && 
+       (node->getOpCodeValue() == TR::lmin || node->getOpCodeValue() == TR::lmax))
+      {
+      return minmaxEvaluator64BitPrimitiveOnIA32(node, cg);
+      }
+
+   TR_X86OpCodes CMP  = BADIA32Op;
+   TR_X86OpCodes MOV  = BADIA32Op;
+   TR_X86OpCodes CMOV = BADIA32Op;
+   switch (node->getOpCodeValue())
+      {
+      case TR::imin:
+         CMP = CMP4RegReg;
+         MOV = MOV4RegReg;
+         CMOV = CMOVG4RegReg;
+         break;
+      case TR::imax:
+         CMP = CMP4RegReg;
+         MOV = MOV4RegReg;
+         CMOV = CMOVL4RegReg;
+         break;
+      case TR::lmin:
+         CMP = CMP8RegReg;
+         MOV = MOV8RegReg;
+         CMOV = CMOVG8RegReg;
+         break;
+      case TR::lmax:
+         CMP = CMP8RegReg;
+         MOV = MOV8RegReg;
+         CMOV = CMOVL8RegReg;
+         break;
+      default:
+         TR_ASSERT(false, "INCORRECT IL OPCODE.");
+         break;
+      }
+   
+   auto oprand0 = cg->evaluate(node->getChild(0));
```

Comment:
I'm not sure if its intentional to abbreviate `operand` by one letter or if its just a spelling mistake, but I think it reads better as `operand`.  Please change throughout this function.

Reply:
It was a typo, not it has been fixed.